src/set_theory/cardinality_number_systems.lean

```diff
@@ -0,0 +1,145 @@
+/-
+Copyright (c) 2019 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn
+
+The cardinality of various number systems, like pnat, int, rat and real.
+-/
+
+import data.real.basic set_theory.ordinal analysis.specific_limits
+
+open nat set
+noncomputable theory
+namespace cardinal
+
+lemma mk_pnat : mk ℕ+ = omega :=
+begin
+  rw [←mk_nat],
+  apply le_antisymm,
+  { apply mk_subtype_le },
+  { fapply mk_le_of_injective succ_pnat_injective }
+end
+
+lemma mk_int : mk ℤ = omega :=
+begin
+  rw [←mk_nat],
+  apply le_antisymm,
+  { have : ∀(k : ℤ), ∃(n : ℕ × ℕ), (n.1 : ℤ) - n.2 = k,
+    { rintro (k|k), use ⟨k, 0⟩, refl, use ⟨0, k.succ⟩, refl },
+    rw [←mul_eq_self (ge_of_eq mk_nat)], apply mk_le_of_surjective this },
+  { fapply mk_le_of_injective nat.cast_injective; apply_instance }
+end
+
+lemma mk_rat : mk ℚ = omega :=
```

Comment:
Can we have these a bit more constructive? It would be nice to have an explicit enumeration using `denumerable Q`, or at least `enumerable Q` if possible.

Reply:
Done in #1104

Reply:
Nice! Shall I put my PR on top of yours, so that I can use your result?